rainforest/runner.go

```diff
@@ -156,3 +156,39 @@ func (c *Client) CheckRunStatus(runID int) (*RunStatus, error) {
 
 	return &runStatus, nil
 }
+
+func (c *Client) LastMatchingRun(params RunParams) (*RunStatus, error) {
```

Comment:
I don't think we need this? Instead in `monitorRunStatus` we can do something like (along with a `max-retries` flag)

```go
if done {
    if status.Result != "passed" {
        remainingRetries = c.Int("max-retries") 
        if remainingRetries > 0 {
            // set run ID for rerun
            os.Setenv("RAINFOREST_RUN_ID", strconv.Itoa(runID))
            // create new context with only decremented max-retries
            flagSet = cli.flag.NewFlagSet("rerun", 0)
            flagSet.Int("max-retries", remainingRetries - 1)
            newContext = cli.NewContext(nil, flagSet)
            return rerunRun(newContext)
        } else {
            return cli.NewExitError("", 1)
        }
}
```